pandas/tslib.pyx

```diff
@@ -656,6 +651,49 @@ for field in fields:
     prop = property(fget=lambda self: np.nan)
     setattr(NaTType, field, prop)
 
+# GH9513 NaT methods (except to_datetime64) to raise, return np.nan, or return NaT
+# create functions that raise, for binding to NaTType
+def _make_error_func(func_name):
+    def f(*args, **kwargs):
+        raise ValueError("NaTType does not support " + func_name)
+    f.__name__ = func_name
+    return f
+
+def _make_nat_func(func_name):
+    def f(*args, **kwargs):
+        return NaT
+    f.__name__ = func_name
+    return f
+
+def _make_nan_func(func_name):
+    def f(*args, **kwargs):
+        return np.nan
+    f.__name__ = func_name
+    return f
+
+_nat_methods = ['date', 'now', 'replace', 'to_datetime', 'today']
+
+_nan_methods = ['weekday', 'isoweekday']
+
+_implemented_methods = ['to_datetime64']
+
+for _method_name in _nat_methods:
+    # not all methods exist in all versions of Python
+    if hasattr(NaTType, _method_name):
+        setattr(NaTType, _method_name, _make_nat_func(_method_name))
+
+for _method_name in _nan_methods:
+    if hasattr(NaTType, _method_name):
+        setattr(NaTType, _method_name, _make_nan_func(_method_name))
+
+for _maybe_method_name in dir(NaTType):
+    _maybe_method = getattr(NaTType, _maybe_method_name)
+    if (callable(_maybe_method)
+        and not _maybe_method_name.startswith("_")
+        and _maybe_method_name not in
+            _implemented_methods + _nat_methods + _nan_methods):
```

Comment:
Actually let me change this line. Having list concatenation in a loop doesn't seem like a good idea.